### ISSUE TITLE
Packages required by examples

### ISSUE THREAD

**Issue by scls19fr** [JuliaGraphics/QML.jl#40]
Hello,

maybe we should find somewhere in doc all packages required by examples.

I personnally had to run:

```julia
Pkg.add("ModernGL")
Pkg.add("GLVisualize")
Pkg.add("GeometryTypes")
Pkg.add("GR")
Pkg.add("TestImages")
Pkg.add("QuartzImageIO")
Pkg.add("Plots")
```

to be able to run all examples.

I'm not sure where these kind of information should be written. But I'm just posting here as it may help some people also (to quickly run examples)

Kind regards

**Comment by barche** [comment]
OK, feel free to add an "Examples" section to the readme and submit a pull request with this list.

**Comment by stemann** [comment]
In addition to the list above, I had to run:
```
Pkg.add("GLAbstraction")
Pkg.add("PyPlot")
Pkg.clone("https://github.com/BenLauwens/StatefulFunctions.jl.git")
```
and modify `example/progressbar_states.jl`.

Errors:
- `gr.jl` fails to plot (on macOS, GR cannot load `QtWidgets.framework/Versions/5/QtWidgets`).
- `image.jl` fails to show image at first click on button (strangely, the second time the button is clicked the image is displayed just fine):
```
$ julia .julia/v0.6/QML/example/image.jl 
MethodError(show, (IOBuffer(data=UInt8[...], readable=true, writable=true, seekable=true, append=false, size=0, maxsize=Inf, ptr=1, mark=-1), MIME type image/png, ColorTypes.RGB{FixedPointNumbers.Normed{UInt8,8}}[RGB{N0f8}(0.886,0.537,0.49) RGB{N0f8}(0.875,0.533,0.51) RGB{N0f8}(0.886,0.522,0.463) ...
```
- `plot.jl` also requires PyPlot (weirdly, GR works fine here).
- `progressbar_states.jl`: StateMachineIterator is now StatefulFunctions - and `@iterator` is now `@stateful`.

**Comment by scls19fr** [comment]
Running exemples could probably be part of unit tests

**Comment by barche** [comment]
OK, thanks for the report. Unfortunately running the examples as an automatic test is not so easy because buttons need to be clicked and the application needs to be quit. Regarding the GR example, are you on the latest GR master, at least this is needed:
https://github.com/jheinen/GR.jl/pull/61

**Comment by ufechner7** [comment]
Fixed in QML 0.8
